app/assets/js/controllers/search-results.js

```diff
@@ -0,0 +1,28 @@
+function SearchResultsController ($stateParams, basketService, productAttributes, searchResponse) {
+	var vm = this;
+
+	vm.query = $stateParams.query;
+
+	vm.results = searchResponse.results;
+
+	vm.addToBasket = basketService.addToBasket;
+
+	vm.productAttributes = productAttributes;
+}
+
+SearchResultsController.resolve = /* @ngInject */ {
+	searchResponse: function($stateParams, searchService, baseSearchParams) {
+		var searchParams = angular.copy(baseSearchParams);
+		searchParams.query = $stateParams.query;
+
+		return searchService
+			.getResults(searchParams)
+			.then(function(response) {
```

Comment:
no, it's not needed. I'd just added it so I could add a breakpoint to the promise resolving. Nice spot :)